packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -64,6 +64,7 @@ export const DotPathReporter: Reporter<string[]> = {
 };
 
 const HEX_STRING_REGEX = /^(0x)?([0-9a-f]{2})+$/gi;
+const DEC_STRING_REGEX = /^[1-9]+[0-9]+([0-9])?$/g;
```

Comment:
I think this regex is wrong. it should be:

```js
const DEC_STRING_REGEX = /^[1-9]+[0-9]*$/g;
```

More importantly, there should be a test that catched this.

Reply:
I totally forgot that we can have 0 balance. I'll apply this:
`^(0|[1-9][0-9]*)$`

Reply:
Oops, that's right, my regex didn't accept 0 :sweat_smile:

Reply:
Same as my initial 😂